src/support/debug.h

```diff
@@ -44,12 +44,6 @@ void setDebugEnabled(const char* types);
 
 #else
 
-// We have an option to build with assertions disabled
-// BYN_ASSERTIONS_ENABLED=OFF, but we currently don't recommend using and we
-// don't test with it.
-#error "binaryen is currently designed to be built with assertions enabled."
```

Comment:
Maybe we should just remove these errors independently of this PR, and then once we have separate CI and Release builds as we've been discussing, then we can run with assertions on CI and without on our release builds.

Reply:
sgtm. @sbc100 you added those specific ones - any concerns? if not i can open a PR.

Reply:
I can't remember exactly why a I added this.  Perhaps we were depending on assert or something like them for normal error reporting.  I'm not sure if that is still true.   I guess now.

Reply:
I assumed it was because we were at one point depending on asserts for error reporting. But we aren't doing that anymore.